### PR TITLE
docs: update Flex diff

### PR DESCRIPTION
**Pull request by zombieJ**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

ref #45255

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    -       |
| 🇨🇳 Chinese |    -       |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 0c169b1</samp>

This pull request updates the documentation of the `Flex` and `Space` components in both English and Chinese. It adds subsections that compare and contrast the two components and their use cases for different layout scenarios.

### <samp>🤖 Generated by Copilot at 0c169b1</samp>

*  Add subsections to the documentation of Flex and Space components to explain the difference and usage scenarios of each component ([link](https://github.com/ant-design/ant-design/pull/45270/files?diff=unified&w=0#diff-36142995ef4bbc8ab96db7fe5ad872e14c4872c2dcfd82634b6f899e1fec2320R17-R21), [link](https://github.com/ant-design/ant-design/pull/45270/files?diff=unified&w=0#diff-1f6b9ec85e9c10788dfb367d525109c02a6430b23bdd94bf2617c5e92f2220e4R18-R22), [link](https://github.com/ant-design/ant-design/pull/45270/files?diff=unified&w=0#diff-abb9d8e13a85136a63d03f3028556dd2948f81a02658caf540818cf1c372c7bbR16-R20), [link](https://github.com/ant-design/ant-design/pull/45270/files?diff=unified&w=0#diff-3386c4be6e559b15e5fb0ee60814a492c147f3527a4eea4beebdf1855dd0c368R20-R24))

### CODE REVIEW

**Review by kiner-tang — APPROVED** [review]

**Review by li-jia-nan — APPROVED** [review]

File: components/flex/index.zh-CN.md [hunk 1]
```diff
@@ -15,6 +15,11 @@ tag: New
 - 适合设置元素之间的间距。
 - 适合设置各种水平、垂直对齐方式。
 
+### 与 Space 组件的区别
```
Comment (crazyair) [inline components/flex/index.zh-CN.md:18]: 这种东西没有复用的写法吗？ cc @PeachScript

Comment (PeachScript) [reply components/flex/index.zh-CN.md:18]: 可以用 embed 复用，正则取部分内容：https://d.umijs.org/guide/markdown#embed

Comment (crazyair) [reply components/flex/index.zh-CN.md:18]: 我试了下 https://github.com/ant-design/ant-design/pull/45280/files，多出来一个何时使用

image

File: components/flex/index.en-US.md [hunk 2]
```diff
@@ -14,6 +14,11 @@ Flex. Available since `5.10.0`.
 - Good for setting spacing between elements.
 - Suitable for setting various horizontal and vertical alignments.
 
+### Difference with Space component
```
Comment (afc163) [inline components/flex/index.en-US.md:17]: 和 Grid 的差异也要写下。

**Pull request merged**